src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java

```diff
@@ -311,11 +312,25 @@ ResultHandle getOutgoingResultHandle() {
     @Override
     public ResultHandle load(String val) {
         Objects.requireNonNull(val);
-        if (val.length() > 65535) {
-            //TODO: we could auto split this, but I don't think we really want strings this long
-            throw new IllegalArgumentException("Cannot load strings larger than " + 65535 + " bytes");
+        int length = val.length();
+        if (length > MAX_STRING_LENGTH) {
+            int div = length / MAX_STRING_LENGTH;
+            int mod = length % MAX_STRING_LENGTH;
+
+            Gizmo.StringBuilderGenerator sb = Gizmo.newStringBuilder(this);
```

Comment:
Minor nit, but shouldn't this allocate one with the correct length, as we know how long the resulting string will be?

Reply:
Good point, I'll update it

Reply:
Done